src/Clerk.ts

```diff
@@ -240,10 +240,20 @@ export default class Clerk {
         }
 
         if (isAuthV2Request(this, headerToken) || isAuthV2Request(this, cookieToken)) {
+          let sessionClaims;
+
+          if (headerToken) {
+            sessionClaims = await verifyToken(this, headerToken);
+          }
+
+          if (!sessionClaims && cookieToken) {
+            sessionClaims = await verifyToken(this, cookieToken);
+          }
```

Comment:
❓ Shall we throw an error in another other scenario?

Reply:
You mean to check in the end if the `sessionClaims` is undefined? Well we can do that and throw a 401 in there. In this way also the user will know that the verification was successfully without the need to check if the `req.sessionClaims` is undefined